contracts/Auction.sol

```diff
@@ -95,20 +89,19 @@ contract Auction is Permissioned {
     external
     auctionNotEnded
     {
-
         euint32 spent = _wfhenix.transferFromEncrypted(msg.sender, address(this), amount);
 
         euint32 newBid = updateHistory(msg.sender, spent);
         // Can't update here highestBid directly because we need and indication whether the highestBid was changed
         // if we will change here the highestBid
         // we will have an edge case when the current bid will be equal to the highestBid
-        euint32 newHeighestBid = FHE.max(newBid, highestBid);
+        euint32 newHighestBid = FHE.max(newBid, highestBid);
+        ebool isNewHighest = newHighestBid.gt(highestBid);
 
-        Eaddress memory eaddr = ConfAddress.toEaddress(payable(msg.sender));
-        ebool wasBidChanged = newHeighestBid.gt(highestBid);
+        eaddress senderEncrypted = FHE.asEaddress(msg.sender);
+        highestBidder = FHE.select(isNewHighest, senderEncrypted, highestBidder);
 
-        highestBidder = ConfAddress.conditionalUpdate(wasBidChanged, highestBidder, eaddr);
```

Comment:
the order of the branches of the `select` was wrong here. this is one of two bugs, the other one being in the FHERC20.sol, where it checked for `le` rather than `lte` so it wouldn't let you send exactly the amount you had in the balance